### PR TITLE
chore(electric): Relax JWT timestamp validation by introducing the idea of "clock drift"

### PR DESCRIPTION
**Pull request by alco**
Fixes VAX-1449, #777.

### CODE REVIEW

File: components/electric/config/runtime.exs [hunk 1]
```diff
@@ -59,7 +59,9 @@ config :electric, Electric.Replication.Postgres,
 
 config :electric, Electric.Postgres.Proxy.Handler.Tracing, colour: true
 
-config :electric, :telemetry_url, "https://checkpoint.electric-sql.com"
+config :electric,
+  max_clock_drift_seconds: 2,
```
Comment (magnetised) [inline components/electric/config/runtime.exs:63]: shouldn't this be configurable? hate to add another env var but...

**Review by magnetised — APPROVED** [review]
if we're going to hard-code 2 seconds as the max drift, then I think it needs some explanation, but otherwise looks great

File: components/electric/config/runtime.exs (hunk 1 above)
Comment (alco) [reply components/electric/config/runtime.exs:63]: It is configurable. One can define a new Docker image based on a publish Electric one, override this `runtime.exs` file and modify any settings that we expose as application env parameters in it. For example:

```
FROM electricsql/electric:0.8.2

RUN echo 'config :electric, max_clock_drift_seconds: N' >> releases/0.8.2/runtime.exs
```

Mapping to an environment variable is a different issue. I don't think it's necessary until we see there's demand for it.

**Comment by davidmartos96** [comment]
@alco I'm assuming this PR removes the need of the following "patch" in the client
https://github.com/electric-sql/electric/blob/4fe5c7f687b4c43e9ba17133645e24179d481aad/clients/typescript/src/auth/secure/index.ts#L17

**Comment by alco** [comment]
> @alco I'm assuming this PR removes the need of the following "patch" in the client

Yes. Thanks for the heads-up!

**Pull request merged**